### ISSUE TITLE
Furthur changes to redesign structures.

### ISSUE THREAD

**Issue by evanlinjin** [bitcoindevkit/bdk#971]
**Describe the enhancement**

After a call with @LLFourn, the follow things will be added (either to this PR, or a separate PR):

1. ~The `PersistBackend<T, C>` trait currently has a generic `T` for the "in memory representation". This also constrains `C` to "accompany" `T`, thus requiring us to have something like `Tracker` in #965. The "onioning" of these structures is unfortunate. However, there is a way to do `PersistBackend` without `T` and just rely on `C`! Let's experiment with that to see if it cleans up the API and reduces "onioning".~
    -- Completed by #965

2. `ChainOracle::get_chain_tip(&self) -> Result<Option<BlockId>, Self::Error>` will be added. This is needed as we need something to input as the "static block" for the `is_block_in_best_chain` call. Additionally, a structure that is capable of implementing `ChainOracle` functionality should also have the functionality to return the tip of the best chain.

3. ~`OwnedIndexer` is bad. Here is something better:~

    ```rust
    pub trait OwnedTxOuts {
        type SpkIndex;
        type Iter: Iterator<Item = (Self::SpkIndex, OutPoint)>;
        fn owned_txouts(&self) -> Self::Iter;
    }
    ```
    ~This thing iterates over all outpoints that we own (alongside the spk index we give it). To get the owned UTXO set, we filter over this. Same thing for balance.~
    -- Turns out `OwnedTxOuts` is not needed either, all removed in #975

**Comment by evanlinjin** [comment]
For point number 3, I think neither the `OwnedIndexer` or `OwnedTxOuts` traits are needed. Will have a PR soon.

**Comment by evanlinjin** [comment]
I don't know whether 2. is actually needed atm.

**Comment by notmandatory** [comment]
Should be fixed in #976

**Comment by notmandatory** [comment]
@evanlinjin can we close this now that #976 is merged?

**Comment by evanlinjin** [comment]
Fixed in #976
